### PR TITLE
disttask: refactor scheduler part

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #46258 

Problem Summary:

### What is changed and how it works?
- rename prev `Scheduler` to `SubtaskExecutor`
- rename prev `SubtaskExecutor` to `MiniTaskExecutor`
- create a new `Scheduler` interface, and each task type should impl it based on `BaseScheduler` which is renamed from previous `InternalScheduler`, and register on framework part.
- create new `Extension` interface to wrap create executor/miniTaskExecutor part, those don't need to be registered now.
- other refactors

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: disttask/framework/scheduler/execute/interface.go
Comment (D3Hunter) [inline disttask/framework/scheduler/execute/interface.go:1]: avoid package cycle

File: disttask/framework/scheduler/execute/interface.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	// MUST return subtask meta back on success.
+	OnFinished(ctx context.Context, subtask []byte) ([]byte, error)
+	// Rollback is used to roll back all subtasks.
+	Rollback(context.Context) error
+}
+
+// MiniTaskExecutor defines the interface of a subtask executor.
+// User should implement this interface to define their own subtask executor.
+// TODO: Rename to minimal task executor.
+type MiniTaskExecutor interface {
```
Comment (tangenta) [inline disttask/framework/scheduler/execute/interface.go:42]: Why not rename it directly?

**Review by tangenta — COMMENTED** [review]
Rest LGTM

**Review by tangenta — APPROVED** [review]

File: disttask/framework/scheduler/execute/interface.go (hunk 1 above)
Comment (D3Hunter) [reply disttask/framework/scheduler/execute/interface.go:42]: will remove the `todo` when we remove `MiniTaskExecutor`

**Review by ywqzzy — APPROVED** [review]

**Pull request merged**